File: app/src/main/java/com/kakacat/minitool/translation/adapter/LanguageAdapter.java
package com.kakacat.minitool.translation.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.kakacat.minitool.R;
import com.kakacat.minitool.common.ui.RecycleViewListener;

import java.util.List;

public class LanguageAdapter extends RecyclerView.Adapter<LanguageAdapter.ViewHolder> {

    private List<String> languageList;
    private RecycleViewListener.OnItemClick listener;

    public LanguageAdapter(List<String> languageList) {
        this.languageList = languageList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.language_item_layout, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.tvLanguage.setText(languageList.get(position));
        if (listener != null) {
            holder.itemView.setOnClickListener(v -> listener.onClick(holder.itemView, position));
        }
    }


    @Override
    public int getItemCount() {
        return languageList.size();
    }

    public void setOnClickListener(RecycleViewListener.OnItemClick listener) {
        this.listener = listener;
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        private TextView tvLanguage;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvLanguage = (TextView) itemView;
        }
    }

}

File: app/src/main/java/com/kakacat/minitool/translation/adapter/LanguageAdapter.kt
package com.kakacat.minitool.translation.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.kakacat.minitool.R
import com.kakacat.minitool.common.ui.RecycleViewListener.OnItemClick

class LanguageAdapter(private val languageList: List<String>) : RecyclerView.Adapter<LanguageAdapter.ViewHolder>() {

    private var listener: OnItemClick? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.language_item_layout, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.tvLanguage.text = languageList[position]
        if (listener != null) {
            holder.itemView.setOnClickListener { listener!!.onClick(holder.itemView, position) }
        }
    }

    override fun getItemCount(): Int {
        return languageList.size
    }

    fun setOnClickListener(listener: OnItemClick?) {
        this.listener = listener
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvLanguage: TextView = itemView as TextView
    }
}